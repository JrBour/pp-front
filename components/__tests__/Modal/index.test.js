import { shallowMount } from '@vue/test-utils'
import Modal from '@/components/Modal'

describe('Modal', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallowMount(Modal, {
      propsData: {
        text: 'Lorem ipsum',
      },
    })
  })

  describe('methods', () => {
    it('should emit cancel method', async () => {
      const cancelButton = wrapper.find('button.modal__action_cancel')
      await cancelButton.trigger('click')
      expect(wrapper.emitted('cancel')).toBeTruthy()
    })
     it('should emit confirm method', async () => {
      const confirmButton = wrapper.find('button.modal__action_confirm')
      await confirmButton.trigger('click')
      expect(wrapper.emitted('confirm')).toBeTruthy()
    })
  })

  describe('snapshot', () => {
    it('renders properly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
