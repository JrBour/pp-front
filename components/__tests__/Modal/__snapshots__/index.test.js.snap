// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal snapshot renders properly 1`] = `
<div class="modal__background">
  <div class="modal__wrapper">
    <p>Lorem ipsum</p>
    <div class="modal__action"><button class="modal__action_cancel">Annuler</button> <button class="modal__action_confirm">
        Confirmer
      </button></div>
  </div>
</div>
`;
